node/src/main/kotlin/net/corda/node/services/statemachine/TransitionExecutorImpl.kt

```
@@ -90,4 +102,25 @@ class TransitionExecutorImpl(
         }
         return Pair(transition.continuation, transition.newState)
     }
+
+    private fun rollbackTransaction() {
```

So this differs slightly from `Action.RollbackTransaction` (as there are no catch blocks at that point). Is idea here that we Handle exceptions thrown in `Action.RollbackTransaction` here?

---

Perhaps this method should be called `rollbackTransactionOnError` to indicate it's not for generic use in any way and to try and convey that it is swallowing/suppressing exceptions during that endeavour.

---

Could do, @williamvigorr3 the point of this rollback is to rollback the transaction when there is a transition error. Errors in user code go to `Action.RollbackTransaction` and executed in the `ActionExecutor`.

I could rename it, but at the same time, I don't like the new name!